tinygrad/runtime/ops_gpu.py

```diff
@@ -9,8 +9,16 @@
 # see test/external/external_osx_profiling.py to determine this ratio. it's in like GPU clocks or something
 OSX_TIMING_RATIO = (125/3) if OSX else 1.0
 
+@functools.lru_cache(maxsize=1)
+def _get_errors():
+  ed = {getattr(cl, k): k for k in filter(lambda e: e.startswith("CL_"), dir(cl)) if getattr(cl, k) <=0}
+  print(ed)
+  return ed
+_ = _get_errors()
 def check(status):
-  if status != 0: raise RuntimeError(f"OpenCL Error {status}")
+  def report(s):
+    raise RuntimeError(f"OpenCL Error {s}: {_get_errors().get(s, 'Unknown error')}")
+  if status != 0: report(status)
```

Comment:
simpler. just `raise RuntimeError`, why `def report` then calling it?

Reply:
Some code of that method was striped to make it cacheable, will throw the error directly.